### PR TITLE
update ci build extending #368

### PR DESCRIPTION
**Pull request by williballenthin**
supersedes #368

i took the nuclear option: i regenerated the pyinstaller spec using the 4.3 release, verified that it produced a working ELF binary, and then copied our configuration into it one-by-one. ultimately, im not actually sure where our problem lies - but it works now.

### CODE REVIEW

**Comment by williballenthin** [comment]
image
this section here passes in an additional parameter `a.datas` that we didn't have, so i wonder if that's why the `encodings` module wasn't available? just a guess, not going to dig any deeper.

**Comment by williballenthin** [comment]
image
w00t

**Comment by mr-tz** [comment]
w00t w00t! I really appreciate you investigating this! This is really cool and I'm relieved it's working now - even without exactly knowing what went wrong.

File: .github/pyinstaller/floss.spec [hunk 1]
```diff
@@ -1,177 +1,83 @@
...
+    upx_exclude=[],
+    runtime_tmpdir=None,
+    console=True,
+)
 
+# enable the following to debug the contents of the .exe
+# writes to ./dist/floss-dat
+coll = COLLECT(
+    exe, a.binaries, a.zipfiles, a.datas, strip=None, upx=True, name="floss-dat"
+)
```
Comment (mr-tz) [inline .github/pyinstaller/floss.spec:83]:
```suggestion
)

```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 name: build
 
 on:
+  # for testing
+  pull_request:
+    branches: [master]
```
Comment (mr-tz) [inline .github/workflows/build.yml:6]: remove before merge once working

**Review by mr-tz — APPROVED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (williballenthin) [reply .github/workflows/build.yml:6]:
```suggestion
```

**Pull request merged**